### PR TITLE
Add search bar styling, PDB Redo filter, auto-suggestions

### PR DESCRIPTION
**Pull request by jvfdunkley**
### :pushpin: References

-   **Issue:** Closes 
 1. [Add filter to show only entries with "PDB-REDO"](https://app.clickup.com/t/245r6y5)
 2. [Redesign filters in covid page](https://app.clickup.com/t/245r5q8)
 3. [Add autosuggestion in covid search](https://app.clickup.com/t/1xprwyv)

### :memo: Implementation
- Implemented new search bar style and filters style. Filters will now be in a mini-menu and filter selections will be chips that are inside the search bar area. 
- Added PDB-Redo as a filter with Arnau's help. Because of the way the PDB-redo info is extracted on the fly, only the rendered rows have PDB-redo info. So I can only filter by PDB-redo on the page that users see. 
- Added autosuggestion and changed the search bar to an Autocomplete Material UI Lab component 

### :art: Screenshots
Screen Shot 2022-02-18 at 12 52 00
Screen Shot 2022-02-18 at 12 51 56
Screen Shot 2022-02-18 at 12 51 52
Screen Shot 2022-02-18 at 12 51 48

### :bookmark_tabs: Others

-   [ ] Any change in the [API repo](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?
-   [ ] Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?

### CODE REVIEW

File: app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomCheckboxFilter.tsx [hunk 1]
```diff
@@ -1,54 +1,95 @@
...
+            <GridMenu
+                open={isOpen}
+                target={anchorEl}
+                onClickAway={closeMenu}
+                position="bottom-start"
+            >
+                <MenuList className="MuiDataGrid-gridMenuList" autoFocusItem={isOpen}>
+                    <MenuItem>
+                        <StyledCheckbox
+                            checked={filterState?.antibodies || false}
```
Comment (tokland) [inline app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomCheckboxFilter.tsx:54]: Maybe at some point `filterState` could be `undefined`, but accordingly to TS, not anymore. So: `filterState.antibodies` should work, right?

File: app/assets/javascripts/covid19/src/webapp/components/structures-table/SearchBar.tsx [hunk 2]
```diff
@@ -1,90 +1,134 @@
...
+                        <li key={data}>
+                            <Chip
+                                label={data.charAt(0).toUpperCase() + data.substr(1).toLowerCase()}
+                                onDelete={handleDelete(data)}
+                                style={styles.chip}
+                            />
+                        </li>
+                    ))}
+                </div>
+                <StyledAutocomplete
```
Comment (tokland) [inline app/assets/javascripts/covid19/src/webapp/components/structures-table/SearchBar.tsx:64]: In the console appears a warning:
```
react_devtools_backend.js:4061 Material-UI: The value provided to Autocomplete is invalid.
None of the options match with `""`.
You can use the `getOptionSelected` prop to customize the equality test.
```

**Review by tokland — COMMENTED** [review]
Code looks very good! I've commited some minor changes to remove a `any` type we had in there, and some other minor details. Some general comments:

- Filter -> onClick only works on the checkbox. Maybe for the full item (checkbox + label)?
- Filter -> when filters are selected/unselected, the search bar width changes, it's a bit confusing. Fixed width? (check with Adrian).
- Pagination: Should set page 1 on text or filter change [this bug was already present in the base code]
- Chip text for renders as "Pdbredo". Maybe PDB-REDO, like in the filter?

**Review by adrianq — CHANGES_REQUESTED** [review]
@jvfdunkley amazing work! On top of Arnau's comments, find below mines:
- Arnau said: "when filters are selected/unselected, the search bar width changes, it's a bit confusing. Fixed width? (check with Adrian)." -> Yes, please set the width of the textfield
- If I type something and them delete it the autosuggestion keeps "Loading" forever:
image
- In general, if I add a string that cannot be found it keeps searching forever (Loading...)
- There are some minor styling things but I will work with @p3rcypj on them

**Review by adrianq — APPROVED** [review]

**Comment by adrianq** [comment]
Thanks @jvfdunkley for the changes! @tokland I am fine with merging this branch. Let me know if there is anything pending on your side. As I said, there are some minor things I would like to refine in terms of styling but I will work with @p3rcypj on them. Once we merge his branch

**Review by tokland — COMMENTED** [review]
The highlighting looks very good! I've committed some changes, let me know if you'd do some of that differently. Only missing the set to page=1 when necessary.

**Review by tokland — APPROVED** [review]
Set-to-page-one is fixed. I'll approve the PR, just a couple of notes for the next iteration:

- Check the warning `None of the options match with `"some search string here"`` in Console.
- If the user enters a search string and then clicks out of the input text, the search is lost. Should it stick?

**Comment by adrianq** [comment]
@jvfdunkley I think we should take care of @tokland second comment at least:
If the user enters a search string and then clicks out of the input text, the search is lost. <- It should stick
I truly think it is important to fix it to improve the user experience. Can you take care of this tomorrow?

**Pull request merged**